core/core-impl/src/main/java/ai/timefold/solver/core/api/score/stream/ConstraintCollectors.java

```diff
@@ -1829,6 +1879,248 @@ public static <A, ResultContainer_, Result_> UniConstraintCollector<A, ResultCon
                 composeFunction);
     }
 
+    // ************************************************************************
+    // consecutive collectors
+    // ************************************************************************
+
+    /**
+     * Creates a constraint collector that returns {@link SequenceChain} about the first fact.
+     *
+     * For instance, {@code [Shift slot=1] [Shift slot=2] [Shift slot=4] [Shift slot=6]}
+     * returns the following information:
+     *
+     * <pre>
+     * {@code
+     * Consecutive Lengths: 2, 1, 1
+     * Break Lengths: 1, 2
+     * Consecutive Items: [[Shift slot=1] [Shift slot=2]], [[Shift slot=4]], [[Shift slot=6]]
+     * }
+     * </pre>
+     *
+     * @param indexMap Maps the fact to its position in the sequence
+     * @param <A> type of the first mapped fact
+     * @return never null
+     */
+    public static <A> UniConstraintCollector<A, ?, SequenceChain<A, Integer>>
+            toConsecutiveSequences(ToIntFunction<A> indexMap) {
+        return InnerUniConstraintCollectors.toConsecutiveSequences(indexMap);
+    }
+
+    /**
+     * Contains info regarding the consecutive sequences and breaks in a collection of points.
+     *
+     * @param <Value_> The type of value in the sequence.
+     * @param <Difference_> The type of difference between values in the sequence.
+     */
+    public interface SequenceChain<Value_, Difference_ extends Comparable<Difference_>> {
```

Comment:
I wonder if these interfaces should be top-level

Reply:
I was actually considering three options:

1. Make them inner interfaces.
2. Put them in the same package as `ConstraintCollectors`. Seemed wrong; these types are not as universal as the other types in that package.
3. Put them in a different package.

Since this was a judgement call and I couldn't really see a big difference between 1 and 3, I decided to go for 1. Seeing as you prefer 1, I made the change.